src/jit/lclvars.cpp

```diff
@@ -670,6 +670,7 @@ void                Compiler::lvaInitUserArgs(InitVarDscInfo *      varDscInfo)
                     {
                     case SystemVClassificationTypeInteger:
                     case SystemVClassificationTypeIntegerReference:
+                    case SystemVClassificationTypeIntegerByReference:
```

Comment:
Here too we can avoid this code duplication by abstracting the code where we need to map struct classification to a type.